pex/platforms.py

```diff
@@ -62,33 +84,43 @@ def create(cls, platform):
 
     @staticmethod
     def _maybe_prefix_abi(impl, version, abi):
+        # type: (str, str, str) -> str
         if impl != "cp":
             return abi
         # N.B. This permits CPython users to pass in simpler extended platform
         # strings like `linux-x86_64-cp-27-mu` vs e.g. `linux-x86_64-cp-27-cp27mu`.
         impl_ver = "".join((impl, version))
         return abi if abi.startswith(impl_ver) else "".join((impl_ver, abi))
 
-    def __new__(cls, platform, impl, version, abi):
-        if not all((platform, impl, version, abi)):
-            raise cls.InvalidPlatformError(
-                "Platform specifiers cannot have blank fields. Given platform={platform!r}, impl={impl!r}, "
-                "version={version!r}, abi={abi!r}".format(
-                    platform=platform, impl=impl, version=version, abi=abi
-                )
-            )
-        platform = platform.replace("-", "_").replace(".", "_")
-        abi = cls._maybe_prefix_abi(impl, version, abi)
-        return super(Platform, cls).__new__(cls, platform, impl, version, abi)
-
     @classmethod
     def from_tags(cls, platform, python, abi):
+        # type: (str, str, str) -> Platform
         """Creates a platform corresponding to wheel compatibility tags.
 
         See: https://www.python.org/dev/peps/pep-0425/#details
         """
         impl, version = python[:2], python[2:]
         return cls(platform=platform, impl=impl, version=version, abi=abi)
 
+    def _tup(self):
+        # type: () -> Tuple[str, str, str, str]
+        return self.platform, self.impl, self.version, self.abi
+
+    def __repr__(self):
+        return "Platform(platform={}, impl={}, version={}, abi={})".format(*self._tup())
+
     def __str__(self):
-        return self.SEP.join(self)
+        # type: () -> str
+        return self.SEP.join(self._tup())
+
+    def __eq__(self, other):
+        # type: (Any) -> bool
+        return self._tup() == other  # type: ignore[no-any-return]
```

Comment:
Idk why MyPy doesn't realize this is a `bool`.

Reply:
A `cast()` might make it more clear what's being done here? This is temporary enough anyway, so not too important.

Reply:
Oh, we can actually indeed use `cast` here because it's a builtin type `bool`! Yay. 

Normally `cast` doesn't work if it's a complex type like `Optional[bool]` because we would need to import `Optional` at runtime.